draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -765,35 +765,64 @@ Otherwise, the processed SD-JWT payload can be passed to the application to be u
 
 # JWS JSON Serialization {#json_serialization}
 
-This section describes an optional alternate format for SD-JWT using the JWS JSON Serialization from [@!RFC7515].
+This section describes an alternative format for SD-JWT using the JWS JSON
+Serialization from [@!RFC7515]. Supporting this format is OPTIONAL.
 
-For both the General and Flattened JSON Serialization, the SD-JWT is represented as a JSON object according
-to Section 7.2 of [@!RFC7515]. The Disclosures (both for issuance and presentation) SHOULD be included in the
-serialized JWS using the member name `disclosures` at the top-level of the JSON object (the same level as the `payload` member). The
-value of the `disclosures` member is an array of strings where each element is an individual Disclosure
-as described in (#creating_disclosures). The Issuer includes a Disclosure for each selectively
-disclosable claim of the SD-JWT payload, whereas the Holder includes only the Disclosures
-selected for the given presentation.
+## New Unprotected Header Parameters {#json_serialization_unprotected_headers}
 
-Alternative methods for conveying the Disclosures MAY be used (such as including them in a `disclosures`
-member of an outer JSON structure also containing the JSON Serialized SD-JWT) as dictated by a specific
-application or transport protocol. However, the details of such approaches fall outside the scope of this
-specification.
+For both the General and Flattened JSON Serialization, the SD-JWT is represented
+as a JSON object according to Section 7.2 of [@!RFC7515]. The following new
+unprotected header parameters are defined:
 
-Verification of the JWS JSON serialized SD-JWT follows the same rules defined in (#verification),
-except that the SD-JWT does not need to be split into component parts and the Disclosures
-can be found in the respective member of the JSON object (or elsewhere).
+ * `disclosures`: An array of strings where each element is an individual
+   Disclosure as described in (#creating_disclosures).
+ * `kb_jwt`: A Key Binding JWT as described in (#kb-jwt).
 
-Using a payload similar to that from [Example 1](#example-1), the following is a non-normative example of
-a JWS JSON serialized SD-JWT from an Issuer with all the respective Disclosures.
+If a Key Binding JWT is present, the digest in the `sd_hash` claim MUST be taken
+over a string built as described in (#integrity-protection-of-the-presentation).
+The "Issuer-signed JWT" part is built by concatenating the protected header, the
+payload, and the signature of the JWS JSON serialized SD-JWT using a `.`
+character as a separator, and using the Disclosures from the `disclosures`
+member of the unprotected header. In case of multiple signatures, only the first
+one is used.
```

Comment:
My first read of this last sentence had me thinking it was only about using the first signature value for constructing the "Issuer-signed JWT" part for input to the `sd_hash`. 

But I think it's also trying to say that, in the case of multiple signatures in the general serialization, the `disclosures` can only show up (and be used for constructing the `sd_hash` input) in the `header` of first of the `signatures` array.  And that the `kb_jwt` has to be there too.

And now I see that is what is said in the General JSON Serialization subsection below. Sorry!

But a bit more clarity here might still be good? I don't know that this is that much better but maybe:

```suggestion
one is used for the Disclosures and Key Binding JWT. 
```

Side note: using only the first one feels kinda icky TBH and the general issue was one reason I'd tried to avoid using the unprotected header for the SD-JWT pieces. But it makes sense to do it this way given all the factors and constraints involved.

Reply:
Thanks, I think the clarification is useful! If we can come up with a better way than just using the first disclosure, I would be open to that.